### DISCUSSION TITLE
single_get/post vs Session

### DISCUSSION THREAD

**Discussion by ohartl** [O-X-L/ansible-opnsense#42] [Q&A]
Is there a reason for both of those to exist? Is it legacy code? @ansibleguy 

I am working on a pull request, where I need to do some work in the api client. I wonder if we could just replace the both `single_get` and `single_post` by its `Session::get` and `Session::post` alternative?

**Comment by ansibleguy** [comment]
Greetings!

I actually use them in some modules that have a simple logic and only need to perform one call, so we do not have to deal with opening and closing the session.

You are right that it can basically be replaced by the session-based one.

I refactored the `single_get` and `single_post` to act as wrapper for the session-based handling: https://github.com/ansibleguy/collection_opnsense/commit/353cef6c3db86e55190818eff19d94f167a829b8

This removes the duplicate logic.

**Reply by ohartl** [reply]
Thanks, this already simplifies things :)

**Reply by ohartl** [reply]
Did create my PR now #46

**Comment by ansibleguy** [comment]
Note: We could also implement a context manager for the session-class so it auto-closes.
Might be even cleaner.

**Reply by ohartl** [reply]
This sounds really good. To be fair, I am not an python expert.
This would definitely make api calls more "fire and forget".

**Reply by ansibleguy** [reply]
Is now also implemented: https://opnsense.ansibleguy.net/en/latest/usage/4_develop.html#api

**Comment by ohartl** [comment]
Btw. I also found `Base::_api_get` to be unused: https://github.com/ansibleguy/collection_opnsense/blob/30512ce94625bb53ad2acdbe6ce493595fd11bff/plugins/module_utils/base/base.py#L628

**Reply by ansibleguy** [reply]
Thanks. I updated all references so they are consistent.
